File: app/src/main/java/it/cammino/risuscito/database/Converters.java
package it.cammino.risuscito.database;

import android.arch.persistence.room.TypeConverter;

import java.sql.Date;

import it.cammino.risuscito.ListaPersonalizzata;

public class Converters {

    @TypeConverter
    public static ListaPersonalizzata fromBlob(byte[] blobAsBytes) {
        return (ListaPersonalizzata) ListaPersonalizzata.Companion.deserializeObject(blobAsBytes);
    }

    @TypeConverter
    public static byte[] fromListaPersonalizzata (ListaPersonalizzata lista) {
        return ListaPersonalizzata.Companion.serializeObject(lista);

    }

    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }

}

File: app/src/main/java/it/cammino/risuscito/database/Converters.kt
package it.cammino.risuscito.database

import androidx.room.TypeConverter
import it.cammino.risuscito.objects.ListaPersonalizzata
import java.sql.Date

@Suppress("unused")
class Converters {

    @TypeConverter
    fun fromBlob(blobAsBytes: ByteArray): ListaPersonalizzata? {
        return ListaPersonalizzata.deserializeObject(blobAsBytes) as ListaPersonalizzata?
    }

    @TypeConverter
    fun fromListaPersonalizzata(lista: ListaPersonalizzata): ByteArray? {
        return ListaPersonalizzata.serializeObject(lista)

    }

    @TypeConverter
    fun fromTimestamp(value: Long?): Date? {
        return if (value == null) null else Date(value)
    }

    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? {
        return date?.time
    }

}
